Rebus.Idempotency/MessageId.cs

```diff
@@ -0,0 +1,53 @@
+namespace Rebus.Idempotency
+{
+    public class MessageId
+    {
+        public string OriginalMessageId;
+        public int DeferCount = 0;
+
+        public MessageId(string messageId, int? deferCount)
```

Comment:
Maybe It is too defensive programming, but we might add some guards on the constructor here. 
* defercount: >  0
* messageId: not nullorwhitespace? It must be lenght of a guid, or we might even use a GUID here or isn't this always the case ?